Mathlib/Init/Logic.lean

```diff
@@ -87,6 +87,27 @@ attribute [refl] Iff.refl
 attribute [trans] Iff.trans
 attribute [symm] Iff.symm
 
+theorem iff_iff_of (h : a) : (a ↔ b) ↔ b :=
+  Iff.intro (fun h₁ => h₁.mp h) (fun h₁ => (iff_true_right h₁).mpr h)
+
+theorem iff_iff_ne_of_ne (h : ¬a) : (a ↔ b) ↔ ¬b :=
+  Iff.intro (fun h₁ => (iff_false_right h).mp h₁.symm) (fun h₁ => (iff_false_right h₁).mpr h)
+
+theorem ne_iff_iff_iff_ne : (¬a ↔ b) ↔ (a ↔ ¬b) := by
+  match Classical.em a with
+  | Or.inl ha => simp only [iff_iff_of ha, iff_iff_ne_of_ne (not_not_intro ha)]
+  | Or.inr ha => simp only [iff_iff_of ha, iff_iff_ne_of_ne ha, Classical.not_not]
+
+theorem iff_assoc : (a ↔ (b ↔ c)) ↔ ((a ↔ b) ↔ c) := by
+  match Classical.em a with
+  | Or.inl ha => simp only [iff_iff_of ha]
+  | Or.inr ha => simp only [iff_iff_ne_of_ne ha, Classical.not_iff]
```

Comment:
The proof is just `tauto`, hopefully?

Reply:
`tauto` is not available here. It is defined in a file that depends on this one. Maybe I can move these theorems somewhere else, but I don't now exactly where it will be a good place

Reply:
Maybe I should move them to `Mathlib/Logic/Basic.lean`?

Reply:
`tauto` also depend on `Mathlib/Logic/Basic.lean`, but I moved it any way for using `iff_not_comm`

Reply:
You should move them to `Logic.Lemmas`. That's exactly the place for "Wants to go in `Logic.Basic` but uses an advanced tactic"